python/ray/data/tests/test_dataset_pipeline.py

```diff
@@ -445,6 +446,69 @@ def test_count_sum_on_infinite_pipeline(ray_start_regular_shared):
     assert 9 == pipe.sum()
 
 
+def test_iter_batches_no_spilling_for_repeat(shutdown_only):
+    # The object store is about 300MB.
+    ctx = ray.init(num_cpus=1, object_store_memory=300e6)
```

Comment:
Could you run this a bunch of times locally to make sure it's not flaky? May be better to run with more memory allowance, and more iterations (e.g., repeat for 10 seconds on a timer, instead of 10 times). This would the test more sensitive to regressions and also probably less flaky.

Reply:
It was consistently passing locally. Changed to run 10 secs instead 10 times.